prober/icmp.go

```diff
@@ -140,42 +140,48 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 			srcIP = net.ParseIP("0.0.0.0")
 		}
 
-		var icmpConn *icmp.PacketConn
-		// If the user has set the don't fragment option we cannot use unprivileged
-		// sockets as it is not possible to set IP header level options.
-		if tryUnprivileged && !module.ICMP.DontFragment {
-			icmpConn, err = icmp.ListenPacket("udp4", srcIP.String())
-			if err != nil {
-				level.Debug(logger).Log("msg", "Unable to do unprivileged listen on socket, will attempt privileged", "err", err)
-			} else {
-				unprivileged = true
-			}
-		}
-
-		if !unprivileged {
-			icmpConn, err = icmp.ListenPacket("ip4:icmp", srcIP.String())
+		if module.ICMP.DontFragment {
+			icmpConn, err := net.ListenPacket("ip4:icmp", srcIP.String())
 			if err != nil {
 				level.Error(logger).Log("msg", "Error listening to socket", "err", err)
 				return
 			}
-		}
 
-		if module.ICMP.DontFragment {
 			rc, err := ipv4.NewRawConn(icmpConn)
 			if err != nil {
 				level.Error(logger).Log("msg", "Error creating raw connection", "err", err)
 				return
 			}
 			socket = &v4Conn{c: rc, df: true}
 		} else {
+			var icmpConn *icmp.PacketConn
+			// If the user has set the don't fragment option we cannot use unprivileged
```

Comment:
Move comment to the other branch of the if, probably update it to say something like "Need to use RawConn rather than ICMP socket in order to set IP header level options."

Reply:
thanks, done!